### ISSUE TITLE
[Color Picker] Ensure canvas element provides a meaningful text equivalent (CAL-AMP52)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#13542]
**monday.com sync:** #10821967470

### Violation: 
 Ensure embedded elements and canvas elements provide a meaningful text equivalent 

### Recommended solution
Add an `aria-label` to the `canvas` element, such as "color swatch" to provide context to assistive technologies:

```html
<canvas aria-label="color swatch" class="color-field" width="238" height="132" style="height: 132px; width: 238px;"></canvas>
```

<img width="1163" height="405" alt="Image" src="https://github.com/user-attachments/assets/dc4fe192-82e0-4bd1-990c-a1af650f7d27" />

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html) 

### Severity: 
 Critical 

### Media Type: 
 Embedded Media and Object Elements 

--- 
### Areas for Remediation: 
- Title: CAL-AMP52 Ensure embedded elements and canvas elements provide a meaningful text equivalent 
- Module: M7 Table and color picker 
- Finding ID: [CAL-AMP52](https://esri.hub.essentia11y.com/short-link/JEPtVXlRuAypoHfS) 

**Issue**
There are meaningful ```<canvas>``` elements with no textual equivalents. Examples include:
- Saturation
- Hue
- Opacity.

**User Impact**
Screen reader users will be unable to determine what this canvas represents.

**Code Reference** 
 ```
<canvas class="color-field" width="297" height="165" style="height: 132px; width: 238px;"></canvas>

```

**Steps to Reproduce**
1. Locate the canvas.
2. Inspect it with Chrome DevTools.
3. Review its contents.
4. In the Accessibility tab, expand the Computed Properties section.
5. Review the value for "Name".
 
**Recommendation**
Ensure embedded elements and canvas elements provide a meaningful text equivalent.
For ```<canvas>``` elements, the best way to provide a textual equivalent is to add child HTML elements to the ```<canvas>``` element. Assistive technologies will access this content instead of the ```<canvas>```.

**Compliant Code Example**
```
<!-- example canvas with short description as the accessible name-->
<canvas id="canvas1" width="200" height="100" aria-label="A traffic light showing green illuminated in the bulb closest to the bottom" role="img"></canvas>
---
<!-- canvas with fallback content as the accessible name -->
<canvas id="canvas1" width="200" height="100"><p>Diagram of the water cycle: Precipitation falls on the ground which merges with ground water which is evaporated into precipitation.</canvas>
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP52.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/JEPtVXlRuAypoHfS) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.
